src/components/InputBar.tsx

```diff
@@ -0,0 +1,81 @@
+import { FC, useEffect, useRef, useState } from 'react';
+import Input from './Input';
+import { PartialChatMessage, PartialNewChatMessage } from '../types';
+import EditBanner from './EditBanner';
+import { makeStyles } from '@material-ui/core/styles';
+import Box from '@material-ui/core/Box';
+import { INITIAL_EDITING_PROPS } from '../constants';
+import { useEditingContext } from '../context/EditingContext';
+
+const useStyles = makeStyles(() => ({
+  wrapper: {
+    width: '100%',
+  },
+}));
+
+type Props = {
+  chatId: string;
+  sendMessageBoxId?: string;
+  sendMessageFunction?: (message: PartialNewChatMessage) => void;
+  editMessageFunction?: (message: PartialChatMessage) => void;
+};
+
+const InputBar: FC<Props> = ({
+  chatId,
+  sendMessageBoxId,
+  sendMessageFunction,
+  editMessageFunction,
+}) => {
+  const classes = useStyles();
+  const { editing, setEditing } = useEditingContext();
```

Comment:
editingProps? Actually you can set plenty of variables in context, so you could have `open`, `body`.. Or define getter `open = editing.open`